### ISSUE TITLE
DETAIL:  Key (history_user_id)=(1) is not present in table "auth_user".

### ISSUE THREAD

**Issue by Gluriurl** [treyhunner/django-simple-history#193]
hi there, i'm trying django-simple-history, it's work properly on my dev env with sqllite but when I run my tests under staging environment that use postgresql i got this error:

IntegrityError: insert or update on table "my_app_historicalfield_type" violates foreign key constraint "my_app_hi_history_user_id_375e8e7412342c93_fk_auth_user_id"
DETAIL:  Key (history_user_id)=(1) is not present in table "auth_user".

thanks !

Anthony

**Comment by macro1** [comment]
Hello @Gluriurl:

I'm guessing your tests generate a user without saving it to the database. Just save the user before using it to perform tracked actions and it should work without any issues.

**Comment by Gluriurl** [comment]
Thanks for your reactivity.
I create the user before with:
user = User.objects.create_user('temporary', '', 'temporary')
..
self.client.login(username='temporary', password='temporary')
..
ModelWithHistory.objects.get_or_create(...)
The error is raised just after that. In my dev environment the very same test is executed without any problem.

Maybe it's due to Postgresql (I have sqlite in dev env) ?

**Comment by macro1** [comment]
@Gluriurl are you also using our `HistoryRequestMiddleware`? That's the only way doing a direct `get_or_create()` would have an idea of which user is logged in.

**Comment by Gluriurl** [comment]
@macro1 Yes i'm using HistoryRequestMiddleware.

**Comment by macro1** [comment]
@Gluriurl: If I have some time I will try to reproduce it, or if you can reproduce it in a minimal project so I could take a look, I might be able to figure out what's going on.

**Comment by dzenkovich** [comment]
I'm having similar problem with my tests. They are functional selenium tests driven by django's manage.py script. I'm getting this `IntegrityError` inside of the second functional test at the very first step when I'm trying to mock data (using factory-boy) of the model that uses simple_history. I've traced problem down to `get_history_user` method and it's call to `thread.request.user` that gives the wrong user. A bit more details in [my question at stackoverflow](stackoverflow.com/questions/36041273/django-selenium-functional-test-simple-history-outdated-history-user)

**Comment by macro1** [comment]
Huh. I can see how that would be troublesome. There's a pull request open to fix exactly this: #213

You can install that branch directly with pip to see if that resolves your issue:
```shell
pip install git+https://github.com/lucaswiman/django-simple-history.git@unset-middleware-request
```

**Comment by macro1** [comment]
The fix in #213 is now merged and released in 1.8.1, please upgrade to that version to avoid the issues with testing while using the middleware.

**Comment by dzenkovich** [comment]
That fixed that, thanks a lot!
